routing/http/server/server.go

```diff
@@ -133,12 +143,28 @@ func Handler(svc ContentRouter, opts ...Option) http.Handler {
 		opt(server)
 	}
 
+	if server.promRegistry == nil {
+		server.promRegistry = prometheus.NewRegistry()
```

Comment:
iiuc this will disable metric for users by default, requiring opt-in via `WithPrometheusRegistry`. 



I noticed we already  call `NewRegistry()` in other places, and changing this requires a slight refactor of tests, so let's keep this as-is and cleanup in  https://github.com/ipfs/boxo/pull/722 without blocking this PR.

Reply:
Yeah. Creating a new registry without access to the registry is an anti-pattern because you can't expose those metrics. 

We should definitely default to the default registry.